### PR TITLE
Update code samples in replication.md for new API

### PR DESCRIPTION
**Pull request by tomblench**
See #410 

Also add para pointing the user to the replication policy documentation.

### CODE REVIEW

File: doc/replication.md [hunk 1]
```diff
@@ -68,65 +66,62 @@ devices. This web service needs to:
...
+synchronised. [Replication policies](replication-policies.md) allow
+for more complex "policies" to be executed, eg "only replicate when on
+Wifi" or "replicate every 30 minutes".
+
 First we create a simple listener that just sets a CountDownLatch when the
 replication finishes so we can wait for a replication to finish without
 needing to poll:
 
 ```java
-import com.cloudant.sync.event.Subscribe;
```
Comment (emlaver) [inline doc/replication.md]: Why were the `import` statements removed?

Comment (tomblench) [inline doc/replication.md]: I can be convinced to add them back but I think they just add noise, and user's IDEs will add them automatically.

I try to think of it this way - the code snipped is not giving a fully formed class which will compile on its own, so why do we need to provide imports?

Comment (emlaver) [inline doc/replication.md]: Ok, makes sense, keep them removed.

File: doc/replication.md [hunk 2]
```diff
@@ -68,65 +66,62 @@ devices. This web service needs to:
...
 
 Replications are set up in code on a device. Use `ReplicatorBuilder` to create
 a pre-configured `Replicator` object. Each `Replicator`
 object can be assigned a `Listener` to receive message when replication
 completes or encounters an error.
 
+The replication examples below are 'one-shot' and will run once until
+completion, at which point the local and remote databases will be
+synchronised. [Replication policies](replication-policies.md) allow
+for more complex "policies" to be executed, eg "only replicate when on
```
Comment (ricellis) [inline doc/replication.md]: as per discussion on the other doc change yesterday, we should change `eg` to `for example`

File: doc/replication.md [hunk 3]
```diff
@@ -68,65 +66,62 @@ devices. This web service needs to:
...
 
-Next we replicate a local datastore to a remote database:
+Next we replicate a local document store to a remote database:
 
 ```java
-import com.cloudant.sync.replication.ReplicatorBuilder;
-import com.cloudant.sync.replication.Replicator;
-
 // Username/password are supplied in the URL and can be Cloudant API keys
 URI uri = new URI("https://username:password@username.cloudant.com/my_database");
```
Comment (ricellis) [inline doc/replication.md]: I'm not sure if "username" is a known account, but this should probably be updated to "examples.cloudant.com" or something in the IANA documentation namespace (although `cloudant.com` probably provides a more clear example than `cloudant.example`).

Comment (ricellis) [inline doc/replication.md]: Note there are multiple instances.

**Comment by tomblench** [comment]
@ricellis fc9cc7481a2f4b67428222d945144efcd92fef5c should address all outstanding changes.

**Pull request merged**